code-challenges.test.js

```diff
@@ -1,56 +1,139 @@
+      expect(takesString(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(takesString(secretCodeWord3)).toEqual("3cc3ntr1c")
+        })
+    })
+
+// // b) Create the function that makes the test pass.
+// //Pseudocode
+// //create a function that called "takesString"
+// // The parameter will be a string and it will converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+// //create a variable called "stringChangetoArr" to store value
+// // use .split method to splits strings into the array. 
+// // use .map to iterate over the array 
+// // use if/else if/else statement to declare the values
+// //return the variable with .join to change the array back to string 
+
+const takesString = (string) => {
+let stringChangetoArr = string.split("").map(value => {
+if (value === "a"){
+    return 4 
+} else if (value === "e" || value === "E"){
+    return 3
+} else if (value === "i"){
+    return 1
+} else if (value === "o"){   
+ return 0
+} else {
+  return value
+}
+}) 
+return stringChangetoArr.join("")
+}
+
+
+// // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
+
+// // a) Create a test with expects statement using the variable provided.
+
+describe("singleLetter",() => {
+
+    it("takes in an array of words and a single letter and returns all the words that contain that particular letter.", () => {
+       
+        const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+        const letterA = "a"
+        // Expected output: ["Apple", "Banana", "Orange"]
+        const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+        const letterE = "e"
+        // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+      expect(singleLetter(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+      expect(singleLetter(arrayOfWords2, letterE )).toEqual(["Cherry", "Blueberry", "Peach"])
+        })
+    })
 
 // b) Create the function that makes the test pass.
-
-
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+//pseudocode
+//takes in an array of words and a single letter and returns all the words that contain that particular letter.
+//create a function called "singleLetter" that takes in an array and a letter
+//create a variable called "bigA" to pass the parameter letter to be also uppercase
+// create a variable called "returnWord" that will filter in the array and will return if the value is true using the method .includes?
+//return value
+
+const singleLetter = (array, letter) => {
+    let bigA = letter.toUpperCase()
+    let returnWord = array.filter(value => {
+    if (value.includes(letter) === true || value.includes(bigA) === true ) {
+        return value
+    }
+    })
+    return returnWord
+    }
```

Comment:
Logic looks great!  Filter has a built-in if statement and so this could be refactored to remove the if statement and just simply return line 93.